test/e2e/storage/vsphere_stress.go

```diff
@@ -0,0 +1,180 @@
+package storage
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/api/core/v1"
+	storageV1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8stype "k8s.io/apimachinery/pkg/types"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"os"
+	"strconv"
+	"sync"
+)
+
+/*
+	Induce stress to create volumes in parallel with multiple threads based on user configurable values for number of threads and iterations per thread.
+	The following actions will be performed as part of this test.
+
+	1. Create Storage Classes of 4 Categories (Default, SC with Non Default Datastore, SC with SPBM Policy, SC with VSAN Storage Capalibilies.)
+    2. READ VCP_STRESS_INSTANCES and VCP_STRESS_ITERATIONS from System Environment.
+	3. Launch goroutine for volume lifecycle operations.
+	4. Each instance of routine iterates for n times, where n is read from system env - VCP_STRESS_ITERATIONS
+	5. Each iteration creates 1 PVC, 1 POD using the provisioned PV, Verify disk is attached to the node, Verify pod can access the volume, delete the pod and finally delete the PVC.
+*/
+var _ = SIGDescribe("vsphere cloud provider stress [Feature:vsphere]", func() {
+	f := framework.NewDefaultFramework("vcp-stress")
+	const (
+		volumesPerNode = 55
+		storageclass1  = "sc-default"
+		storageclass2  = "sc-vsan"
+		storageclass3  = "sc-spbm"
+		storageclass4  = "sc-user-specified-ds"
+	)
+	var (
+		client     clientset.Interface
+		namespace  string
+		instances  int
+		iterations int
+		err        error
+	)
+
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("vsphere")
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+
+		nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		Expect(nodeList.Items).NotTo(BeEmpty(), "Unable to find ready and schedulable Node")
+
+		// if VCP_STRESS_INSTANCES = 12 and VCP_STRESS_ITERATIONS is 10. 12 threads will run in parallel for 10 times.
+		// Resulting 120 Volumes and POD Creation. Volumes will be provisioned with each different types of Storage Class,
+		// Each iteration creates PVC, verify PV is provisioned, then creates a pod, verify volume is attached to the node, and then delete the pod and delete pvc.
+
+		Expect(os.Getenv("VCP_STRESS_INSTANCES")).NotTo(BeEmpty(), "ENV VCP_STRESS_INSTANCES is not set")
+		instances, err = strconv.Atoi(os.Getenv("VCP_STRESS_INSTANCES"))
```

Comment:
I'd reverse these two lines (and all similar) so that you don't call `os.Getenv` twice per env variable:
```
instancesStr := os.Getenv("VCP_STRESS_INSTANCES")
Expect(instanceStr).NotTo(BeEmpty(), "ENV VCP_STRESS_INSTANCES is not set")
instances, err := strconv.Atoi(instanceStr)
Expect(....)
```

Reply:
fixed.